### PR TITLE
Feat/controller implementation

### PR DESCRIPTION
**Pull request by Courtcircuits**
Implementation of the controller

### CODE REVIEW

File: .idea/.gitignore
Comment (sameo) [inline .idea/.gitignore:1]: Please remove all `.idea`files.

File: controller/src/action/action_repository.rs
Comment (hugoponthieu) [inline controller/src/action/action_repository.rs:10]: What is container_uri ?

**Review by hugoponthieu — CHANGES_REQUESTED** [review]
I was wondering if adding route for fetching a stream that expose the state of the pipeline was possible

File: docs/arch/lexicon.md
Comment (theotchlx) [inline docs/arch/lexicon.md:10]: Please re-add this removed empty line at the end of this .md file

No blank lines at the end of a file is not up to markdown standards,

And generally, it is best to leave one blank line at the end of any file (source code included), as many older tools won't work properly if a line feed isn't present. Also it is a POSIX standard, as it is the definition of a line: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206

**Review by theotchlx — CHANGES_REQUESTED** [review]
Hi, thank you for this great work and PR.

I suggested below some adjustments and questions

File: controller/src/scheduler/mod.rs
Comment (Courtcircuits) [reply controller/src/scheduler/mod.rs:56]: You can put code inside `mod.rs`. E.g : https://github.com/tokio-rs/axum/blob/main/axum/src/response/mod.rs

File: controller/src/pipeline/pipeline_service.rs
Comment (Courtcircuits) [reply controller/src/pipeline/pipeline_service.rs:113]: Nope we will get rid of it

File: controller/src/pipeline/pipeline_controller.rs
Comment (Courtcircuits) [reply controller/src/pipeline/pipeline_controller.rs:74]: Yes

File: controller/.gitignore
Comment (Courtcircuits) [reply controller/.gitignore:1]: We'll do it

**Review by hugoponthieu — APPROVED** [review]
The code is clean, consider merging this part

**Comment by Courtcircuits** [comment]
> The code is clean, consider merging this part

We still need to separate the "commands" into a dedicated repository. This is in progress, so until it's committed I would rather not merge this branch. Sorry for the wait

**Pull request merged**
